storage/innobase/dict/dict0dict.cc

```diff
@@ -1663,23 +1663,36 @@ dict_table_rename_in_cache(
 			       table->name.m_name);
 		}
 		dict_mem_foreign_table_name_lookup_set(foreign, do_alloc);
-		if (strchr(foreign->id, '/')) {
+		/* All table names are internally stored in charset
+		my_charset_filename (except the temp tables and the
+		partition identifier suffix in partition tables). The
+		foreign key constraint names are internally stored
+		in UTF-8 charset.  The variable fkid here is used
+		to store foreign key constraint name in charset
+		my_charset_filename for comparison further below. */
+		char    fkid[MAX_TABLE_NAME_LEN * 2 + 20];
+
+		if (const char* s= strchr(foreign->id, '\377')) {
```

Comment:
We do not really need the `fkid[]` buffer here, and we should invoke `foreign->sql_id()` to extract the intended visible part of the constraint name, just like we do everywhere else.